src/sentry/integrations/msteams/webhook.py

```diff
@@ -180,3 +199,60 @@ def handle_member_removed(self, request):
 
         integration.delete()
         return self.respond(status=204)
+
+    def handle_action_submitted(self, request):
+        data = request.data
+        channel_data = data["channelData"]
+        team_id = channel_data["team"]["id"]
+        tenant_id = channel_data["tenant"]["id"]
+        group_id = data["value"]["groupId"]
+        user_id = data["from"]["id"]
```

Comment:
Will all action events contain all of these keys or should you capture and log KeyError here?

Reply:
To my knowledge, any Activity we get from teams _should_ include the `team` and `tenant` fields. Whether or not they always _do_ is not necessarily known. I suppose it would be possible for Microsoft to mess that up. Similarly, an Activity with `message` type should have a `from` field, but see previous point. The `value` field and the `groupId` within it are checked for before entering this function.

Reply:
We should have all of these at this point, but if we come across a situation where for some reason we don't have any of these, we'd rather it just error out and show up in Sentry so we can see it and handle that case.